test/test_eos_vinet.cpp

```diff
@@ -0,0 +1,488 @@
+//                        "Temperature");
+//        }
+//      }
+
+//      WHEN("A B_T(rho, T) lookup is performed") {
+//        eos.TBulkModulusFromDensityTemperature(v_density, v_temperature, v_tbulkmodulus, num);
+//#ifdef PORTABILITY_STRATEGY_KOKKOS
+//        Kokkos::fence();
+//        Kokkos::deep_copy(h_tbulkmodulus, v_tbulkmodulus);
+//#endif // PORTABILITY_STRATEGY_KOKKOS
+//        THEN("The returned B_T(rho, T) should be equal to the true value") {
+//          array_compare(num, density, temperature, h_tbulkmodulus, tbulkmodulus_true, "Density",
+//                        "Temperature");
+//        }
+//      }
+
+      WHEN("A Cv(rho, T) lookup is performed") {
+        eos.SpecificHeatFromDensityTemperature(v_density, v_temperature, v_cv, num);
+#ifdef PORTABILITY_STRATEGY_KOKKOS
+        Kokkos::fence();
+        Kokkos::deep_copy(h_cv, v_cv);
+#endif // PORTABILITY_STRATEGY_KOKKOS
+        THEN("The returned T(rho, e) should be equal to the true value") {
+          array_compare(num, density, temperature, h_cv, cv_true, "Density",
+                        "Temperature");
+        }
+      }
+
+//      WHEN("A alpha(rho, T) lookup is performed") {
+//        eos.TExpansionCoeffFromDensityTemperature(v_density, v_temperature, v_alpha
+//                                                    num);
+//#ifdef PORTABILITY_STRATEGY_KOKKOS
+//        Kokkos::fence();
+//        Kokkos::deep_copy(h_alpha, v_alpha);
+//#endif // PORTABILITY_STRATEGY_KOKKOS
+//        THEN("The returned alpha(rho, T) should be equal to the true value") {
+//          array_compare(num, density, temperature, h_alpha, alpha_true, "Density",
+//                        "Temperature");
+//        }
+//      }
+    }
+  }
+}
+SCENARIO("Vinet EOS SetUp", "[VectorEOS][VinetEOS]") {
+  GIVEN("Parameters for a Vinet EOS") {
+    // Unit conversions
+    constexpr Real cm = 1.;
+    constexpr Real us = 1e-06;
+    constexpr Real Mbcc_per_g = 1e12;
+    // Vinet parameters for copper
+    Real rho0 = 8.93;
+    Real T0 = 298.0;
+    Real B0 = 1.3448466 * Mbcc_per_g;
+    Real BP0 = 4.956;
+    Real A0 = 5.19245e-05;
+    Real Cv0 = 0.383e-05 * Mbcc_per_g;
+    constexpr Real E0 = 0.0;
+    constexpr Real S0 = 5.05e-04 * Mbcc_per_g;
+    constexpr Real d2to40[39]={0.,0.,0.0000000000001,0.,0.,0.,0.,0.,0.,0.,
+            0.,0.,0.,0.,0.,0.,0.,0.,0.,0.,
+            0.,0.,0.,0.,0.,0.,0.,0.,0.,0.,
+            0.,0.,0.,0.,1.0e-26,0.,0.,0.,0.};
+    // Create the EOS
+    EOS host_eos = Vinet(rho0, T0, B0, BP0, A0, Cv0, E0, S0, d2to40);
+    EOS eos = host_eos.GetOnDevice();
+
+    eos.PrintParams();
+
+    eos.Finalize();
+
+    WHEN("Faulty/not set parameter rho0 is given") {
+      rho0 = -1.0;	    
+      host_eos = Vinet(rho0, T0, B0, BP0, A0, Cv0, E0, S0, d2to40); 
+      eos = host_eos.GetOnDevice();
+      eos.PrintParams();
+      eos.Finalize();
+      THEN("An error message should be written out") {
+
+      }
```

Comment:
Did we figure out how to test for the errors here?

Reply:
No

Reply:
I will try to look at fixing this particular issue when I get the chance later in this week maybe tomorrow. I think the right solution involves cmake. 

I didn't read the tests carefully yet.